cmd/snap/cmd_add_skill_test.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"fmt"
+	"net/http"
+
+	. "gopkg.in/check.v1"
+
+	. "github.com/ubuntu-core/snappy/cmd/snap"
+)
+
+func (s *SnapSuite) TestAddSkillHelp(c *C) {
```

Comment:
I have mixed feelings about this kind of test, I think we are testing too much the whimsy of go-flags but maybe is what we want,  I would just test that the help contains the bit we care about, basically only:

+[add-skill command options]
 +      -a=          key=value attributes
 +          --app=   list of apps providing this skill
 +          --label= human-friendly label
 +
 +[add-skill command arguments]
 +  snap:            name of the snap offering the skill
 +  name:            skill name within the snap
 +  type:            skill type

the rest in either clearly readable in the code or is mechanically produced or related to this by go-flags